bayes_opt.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy.optimize import minimize
+from sklearn.gaussian_process import GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import Matern
+
+
+class BayesianOptimizer(object):
+
+    def __init__(self, feature_meta,
+                 init_observations=[],
+                 kernel=Matern(nu=.5)):
+        self.feature_names = feature_meta.keys()
+        self.feature_bounds = np.stack(feature_meta.values())
+        self.features_dim = len(self.feature_names)
+        self.observations = init_observations
+        self.i = 0
+        self.kernel = Matern(nu=.5)
```

Comment:
Heh, you should probably set this to "kernel" so you don't go crazy trying to figure out why the kernel doesn't change when you pass a new one to the constructor....